shared/src/main/kotlin/org/javacs/kt/storage/Storage.kt

```diff
@@ -0,0 +1,73 @@
+package org.javacs.kt.storage
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.builtins.SetSerializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+import org.javacs.kt.LOG
+import java.lang.Exception
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+class Storage(val path: Path) {
+    fun getSlice(slicePath: String): Storage? {
+        val fullSlicePath = Paths.get(path.toString(), slicePath)
+
+        if (!Files.exists(fullSlicePath)) {
+            Files.createDirectory(fullSlicePath)
+        } else if (!Files.isDirectory(fullSlicePath)) {
+            return null
+        }
```

Comment:
I think we could simplify this to something along the lines of

```suggestion
        Files.createDirectories(fullSlicePath)
```

and catch the `FileAlreadyExistsException` in case it's a file rather than a directory (since I'd assume that happens rarely and callers generally use correct paths)?